helix-lock/src/test/java/org/apache/helix/lock/helix/TestZKHelixNonblockingLockWithPriority.java

```diff
@@ -81,6 +81,45 @@ public void afterSuite() throws IOException {
     super.afterSuite();
   }
 
+  @Test
```

Comment:
not sure if we want to add a test case, where our user are using the old default priority (-1) for creating the lock and then re-acquire the lock or releae the lock, will it work? Essentially, we need to fix for customers who have been using with broken code also.

Reply:
Basically, customers will never have a lock that is recorded with priority = -1. 

When creating the `ZKDistributedNonblockingLock`, there is [check](https://github.com/apache/helix/blob/master/helix-lock/src/main/java/org/apache/helix/lock/helix/ZKDistributedNonblockingLock.java#L134) to make sure the priority is  not negative. The priority = -1 can only be seen if the field `priority` is not represent in the lock and we need to read it by [`getPriority()`](https://github.com/apache/helix/blob/master/helix-lock/src/main/java/org/apache/helix/lock/LockInfo.java#L173). Because the field `priority` is missing, `getPriority()` will return old default priority -1 (now changed to 0) and [updater](https://github.com/apache/helix/blob/master/helix-lock/src/main/java/org/apache/helix/lock/helix/ZKDistributedNonblockingLock.java#L316) in this way uses -1  to determine whether to lock or unlock.

Hope this explains.